### PR TITLE
Rename misleading instance variable names in FeedbackResponseAttributes #5456

### LINKED ISSUE #5456: Rename misleading instance variable names in FeedbackResponseAttributes
Instance variables `recipientEmail` and `giverEmail` may not always be emails.
Depending on the question type, they may be team names etc.

### PR DESCRIPTION
**Pull request by YongJieYongJie**
Fixes #5456

### CODE REVIEW

File: src/main/java/teammates/logic/core/FeedbackSessionsLogic.java [hunk 1]
```diff
@@ -894,24 +894,24 @@ private StringBuilder getFeedbackSessionResultsForQuestionInCsvFormat(
...
             // print missing responses from the current giver
             if (isNewGiver) {
                 exportBuilder.append(getRowsOfPossibleRecipientsInCsvFormat(fsrBundle,
                         question, questionDetails,
                         possibleRecipientsForGiver, prevGiver));
                 String giverIdentifier = question.giverType == FeedbackParticipantType.TEAMS
-                                       ? fsrBundle.getFullNameFromRoster(response.giverEmail)
-                                       : response.giverEmail;
+                                       ? fsrBundle.getFullNameFromRoster(response.giver)
+                                       : response.giver;
```
Comment (chowyb) [inline src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: Maybe indent these 2 lines 2 spaces to the right as well? It seems to be the convention now that `?` and `:` are not aligned with `=` for assignment.

File: src/test/java/teammates/test/cases/logic/StudentsLogicTest.java [hunk 2]
```diff
@@ -411,13 +411,13 @@ public void testAdjustFeedbackResponseForEnrollments() throws Exception {
                                             Integer.parseInt(feedbackResponse1InBundle.feedbackQuestionId));
         FeedbackResponseAttributes responseBefore =
                 frLogic.getFeedbackResponse(feedbackQuestionInDb.getId(),
-                                            feedbackResponse1InBundle.giverEmail,
-                                            feedbackResponse1InBundle.recipientEmail);
+                                             feedbackResponse1InBundle.giver,
+                                             feedbackResponse1InBundle.recipient);
```
Comment (chowyb) [inline src/test/java/teammates/test/cases/logic/StudentsLogicTest.java]: These lines seem to have been shifted to the right by 1 space, probably unintended.

File: src/test/java/teammates/test/driver/TestProperties.java [hunk 3]
```diff
@@ -81,6 +84,9 @@
             BROWSER = prop.getProperty("test.selenium.browser");
             FIREFOX_PATH = prop.getProperty("test.firefox.path");
             
+            IS_HEADLESS_FIREFOX = Boolean.parseBoolean(prop.getProperty("test.is_headless_firefox"));
+            XVFB_PORT = prop.getProperty("test.xvfb_port");
+
```
Comment (chowyb) [inline src/test/java/teammates/test/driver/TestProperties.java]: Not sure if these changes are needed

File: src/test/java/teammates/test/pageobjects/Browser.java [hunk 4]
```diff
@@ -83,8 +84,16 @@ private WebDriver createWebDriver() {
...
 
+            if (TestProperties.IS_HEADLESS_FIREFOX) {
+                System.out.println("Running Firefox is headless mode with Xvfb on display port "
+                                   + TestProperties.XVFB_PORT
+                                   + " (Xvfb must already be listening on that port)");
+                FirefoxBinary firefoxBinary = new FirefoxBinary(new File(firefoxPath));
+                firefoxBinary.setEnvironmentProperty("DISPLAY", TestProperties.XVFB_PORT);
+                return new FirefoxDriver(firefoxBinary, profile);
+            }
+            return new FirefoxDriver(profile);
```
Comment (chowyb) [inline src/test/java/teammates/test/pageobjects/Browser.java]: Same here, not sure if these changes are needed

File: src/test/resources/data/StudentFeedbackSubmissionEditPageActionTest.json [hunk 5]
```diff
@@ -1456,4 +1456,4 @@
         }
         
     }
-}
\ No newline at end of file
+}
```
Comment (chowyb) [inline src/test/resources/data/StudentFeedbackSubmissionEditPageActionTest.json]: Quite a number of these additional newlines at the end of the file, but I don't think it really hurts the PR so it should be okay.

**Comment by chowyb** [comment]
Some minor changes requested, as well as the removal of your headless browser experimentation (pretty sure you didn't plan on pushing that :stuck_out_tongue:)

File: src/test/java/teammates/test/driver/TestProperties.java (hunk 3 above)
Comment (YongJieYongJie) [inline src/test/java/teammates/test/driver/TestProperties.java]: oops, that's my owning testing configuration

File: src/test/resources/data/StudentFeedbackSubmissionEditPageActionTest.json (hunk 5 above)
Comment (YongJieYongJie) [inline src/test/resources/data/StudentFeedbackSubmissionEditPageActionTest.json]: Actually I don't know why this is showing up, I checked the files and there are no new empty lines at the end of the files.

**Comment by YongJieYongJie** [comment]
@chowyb Ready for re-review!
Fixed the indentation and removed my custom testing configurations (was just to irritating to have Firefox popping up when I'm trying to do work while tests run)

**Comment by chowyb** [comment]
@YongJieYongJie alright, looks good now :+1:

**Comment by damithc** [comment]
@kanghj can you have a look too? I'm not sure if there are hidden problems we missed.

**Comment by kanghj** [comment]
Looks good to me. @YongJieYongJie, I would like to request for one more possibly related change though, if you don't mind, in `FeedbackResponse.java`, can you remove the TODO on `private String receiver;`, `//TODO: rename to receiverEmail, will require database conversion`?

**Comment by YongJieYongJie** [comment]
@kanghj Removing `// TODO` is always good :thumbsup:

**Comment by YongJieYongJie** [comment]
@kanghj Then do we need to attach another `// TODO` (or create an issue) to rename `giverEmail` to `giver` in `FeedbackResponse`?

**Comment by kanghj** [comment]
Yes, let's make an issue to rename it.

**Comment by YongJieYongJie** [comment]
@kanghj Issue created: 
Rename giverEmail to giver in ResponseFeedback #5571

@damithc Ready for final review!

**Pull request merged**